src/components/encounter/encounter-form-manager.ts

```diff
@@ -78,6 +79,26 @@ export class EncounterFormManager {
     return encounterForSubmission;
   }
 
+  static prepareProgramEnrollment(fields: FormField[], encounterDate: Date, encounterLocation: string, patient: fhir.Patient) {
+    const patientProgramWorkflow = fields.find((field) => field.type == 'programWorkflow');
+    const patientProgramState = fields.find((field) => field.type == 'programState')?.value;
+    const completionDate = fields.find((field) => field.questionOptions.isProgramCompletion === true)?.value;
+    const programUuid = patientProgramWorkflow?.questionOptions?.datasource?.config;
+  
```

Comment:
Is it possible to have more than one field with type `programWorkflow` or `programState` or `completionDate` within a form? If so, how would this implementation handle it?

Reply:
The completion date is always one ,  but there is a likelyhood of having more than one `programState` or `programWorkflow`